extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -385,33 +398,36 @@ export class AmpStory360 extends AMP.BaseElement {
   }
 
   /**
-   * Creates a device orientation listener and sets gyroscopeControls_ state.
-   * If listener is not called in 1000ms, remove listener and resume animation.
-   * This happens on desktop browsers that support deviceorientation but aren't in motion.
+   * Listens for deviceorientation events.
+   *
+   * Some browsers support the 'deviceorientation' event but never call it.
+   * This waits for one call before initiating a constant listener.
    * @private
    */
   enableGyroscope_() {
-    this.gyroscopeControls_ = true;
-    this.togglePermissionClass_(true);
-
-    const checkNoMotion = this.timer_.delay(() => {
-      this.gyroscopeControls_ = false;
-      if (this.isReady_ && this.isPlaying_) {
-        this.animate_();
-      }
-    }, 1000);
-
-    let rafTimeout;
-
-    this.win.addEventListener('deviceorientation', (e) => {
-      if (this.isReady_ && this.isOnActivePage_) {
-        // Debounce onDeviceOrientation_ to rAF.
-        rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
-        rafTimeout = this.win.requestAnimationFrame(() =>
-          this.onDeviceOrientation_(e)
-        );
+    if (this.element.getAttribute('controls') !== 'gyroscope') {
+      return;
+    }
+    // Listen for one call before initiating.
+    listenOncePromise(this.win, 'deviceorientation').then(() => {
+      this.gyroscopeControls_ = true;
+      this.togglePermissionClass_(true);
+      // Debounce onDeviceOrientation_ to rAF.
+      let rafTimeout;
+      this.win.addEventListener('deviceorientation', (e) => {
+        if (this.isReady_ && this.isOnActivePage_) {
+          rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
+          rafTimeout = this.win.requestAnimationFrame(() =>
+            this.onDeviceOrientation_(e)
+          );
+        }
+      });
+      // Display discovery animation.
+      if (this.isOnActivePage_) {
+        const storyEl = this.win.document.querySelector('amp-story');
```

Comment:
Why did you move away from having it as part of the component? I think it's weird to have it at a random position in the DOM, especially as we can create many of them and never do clean up.

Reply:
It's the same problem as the activate button #30420. 
Content overlaps it since it's a child of `amp-story-grid-layer`.

I converted this to draft since this needs work.